### PR TITLE
Worker: don't blow up if a run doesn't have a start node

### LINKED ISSUE #758: Worker: bad crash if a workflow is invalid
Still investigating this

If the worker receives a bad  workflow (basically no start node), it'll blow up very early in the execute pipeline. Probably before we have good guards.

image

I'm not quite sure what the consequences of this crash are. Am investigating. Will add guard code.

The problem is occurring because Lightning is sending us "bad" jobs, likely as a result of sync issues

### PR DESCRIPTION
**Pull request by josephjclark**
We're seeing an error where the worker will blow up if a run does not have a start node (or specifies a start node which does not exist).

This causes the worker to immediately exit and all outstanding runs will be lost. Not a great look.

Fixes #758 

The fix is easy enough and I've added unit tests. The runtime will catch the inconsistent workflow structure and raise an exception for us - we just need the worker to not blow up before this happens.

There are a few other areas around this code where an exception will blow up the worker. I've raised an issue to track them #759  but I don't think I want to take them on here. Let's just fix the immediate issue.

### CODE REVIEW

**Comment by josephjclark** [comment]
Kinda blocked by #752 - need to know I can merge safely

**Review by taylordowns2000 — APPROVED** [review]
looks good. thank you. can be merged after rebase

**Comment by josephjclark** [comment]
@taylordowns2000 worker 1.6.1 is built and the PR is merged

**Pull request merged**
